tests/test_simple_pipeline_executor.py

```diff
@@ -243,14 +237,21 @@ def test_from_pipeline_output_differ(self):
             )
         )
 
+        executor = self._configure_pipeline(
+            NoDimensionsTestTask.ConfigClass,
+            NoDimensionsTestTask.ConfigClass,
+            storageClass_a="TaskMetadataLike",
+        )
+
         with self.assertLogs("lsst", level="INFO") as cm:
             quanta = executor.run(register_dataset_types=True, save_versions=False)
-        # a has been told to return a TaskMetadata but will convert to dict.
+        # a has been told to return a TaskMetadata but this will convert to
+        # dict on read by b.
         # b returns a dict and that is converted to TaskMetadata on put.
         self._test_logs(cm.output, "dict", "lsst.pipe.base.TaskMetadata", "dict", "dict")
```

Comment:
This is saying:

The test in line 250 is saying that

* "a" was given a dict
* "a" returned a TaskMetadata
* "b" was given a dict
* "b" returned a dict

which implies that "intermediate" should be a TaskMetadata and output should be a dict?

Reply:
The log messages are all from the perspective of the task, and are consistent with the task's connections.  They don't show the conversions that happen at the end of `put` or the beginning of `get`.